src/Diagnostics.RuntimeHost/Services/HealthCheckService/HealthCheckService.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+
+namespace Diagnostics.RuntimeHost.Services
+{
+    public interface IHealthCheckService : IDisposable
+    {
+        Task RunHealthCheck(bool retry);
+    }
+
+    public class HealthCheckService : IHealthCheckService
+    {
+        private string OutboundConnectivityCheckUrl;
+        private HttpClient _httpClient;
+        IConfiguration _configuration;
+        bool IsOutboundConnectivityCheckEnabled = false;
+
+        public HealthCheckService(IConfiguration Configuration)
+        {
+            _configuration = Configuration;
+            IsOutboundConnectivityCheckEnabled = Convert.ToBoolean(_configuration["HealthCheckSettings:IsOutboundConnectivityCheckEnabled"]);
+            if (IsOutboundConnectivityCheckEnabled)
+            {
+                OutboundConnectivityCheckUrl = _configuration["HealthCheckSettings:OutboundConnectivityCheckUrl"];
+                if (OutboundConnectivityCheckUrl != null)
+                InitializeHttpClient();
+            }
+        }
+
+        private Task<HttpResponseMessage> Get(HttpRequestMessage request)
+        {
+            return _httpClient.SendAsync(request);
+        }
+
+        public async Task RunHealthCheck(bool retry)
+        {
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, OutboundConnectivityCheckUrl);
+            try
+            {
+                var response = await Get(request);
+            }
+            catch (Exception ex)
+            {
+                if (retry)
```

Comment:
We wrote a RetryAsync  utility sometime back.  we can prob use that as well.

https://github.com/Azure/Azure-AppServices-Diagnostics/blob/7b4bd51ad8b0c228e2c7a85421e755714c328556/src/Diagnostics.RuntimeHost/Utilities/RetryHelper.cs#L10